src/classes/BDI_MigrationMappingHelper.cls

```diff
@@ -37,125 +37,113 @@
+                FROM Data_Import_Field_Mapping_Set__mdt
+                WHERE DeveloperName = :BDI_MigrationMappingUtility.MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME];
+
+            return migratedFieldMappingSetCount > 0 ? true : false;
+        }
+    }
 
-        Map<String, BDI_HelpTextCMT.MappingDetail> helpTextFieldMap =
-                new Map<String, BDI_HelpTextCMT.MappingDetail>();
+    /*******************************************************************************************************
+    * @description Default packaged Object Mapping Set metadata record
+    */
+    public Data_Import_Object_Mapping_Set__mdt getDefaultObjectMappingSet() {
+        return [SELECT Id, DeveloperName, Label
+            FROM Data_Import_Object_Mapping_Set__mdt
+            WHERE DeveloperName = :BDI_MigrationMappingUtility.DEFAULT_OBJECT_MAPPING_SET_RECORD_NAME];
+    }
 
-        Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts> importStatusAndLookupMap =
-                new Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts>();
+    /*******************************************************************************************************
+    * @description List of bad/malformed help text mappings
+    */
+    public List<BDI_MigrationMappingUtility.HelpTextFieldMapping > badHelpTextFieldMappings =
+        new List<BDI_MigrationMappingUtility.HelpTextFieldMapping >();
+
+    /*******************************************************************************************************
+    * @description List of BDI_FieldMappingHelpText.DataImportFieldMappings built from good help text mappings
+    */
+    public List<BDI_MigrationMappingUtility.DataImportFieldMapping> dataImportFieldMappings =
+        new List<BDI_MigrationMappingUtility.DataImportFieldMapping>();
+
+    /*******************************************************************************************************
+     * @description Go through all fields in Data Import object and populate relevant lists from help text mappings
+     */
+    public void populateListsFromHelpTextMappings() {
 
-        Map<String, Schema.DescribeFieldResult> allDFRs =
-                UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.StrAllNSPrefix('DataImport__c'));
+        badHelpTextFieldMappings.clear();
+        dataImportFieldMappings.clear();
 
+        Map<String, Schema.DescribeFieldResult> describeFieldResultsByFieldName =
+            UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.StrAllNSPrefix(
+                BDI_MigrationMappingUtility.DATA_IMPORT_OBJECT_NAME));
 
-        for (String dataImportFieldName : allDFRs.keySet()) {
+        for (String dataImportFieldName : describeFieldResultsByFieldName.keySet()) {
 
-            // check if field is importstatus or imported
             Boolean isImportStatus = dataImportFieldName.contains('importstatus');
             Boolean isImportLookup = dataImportFieldName.contains('imported');
```

Comment:
The code should probably check against a specific list of fields here since customers will be able to map from their own newly created custom fields on DataImport__c and could potentially give their new fields names that include these strings...